src/Parser.php

```diff
@@ -12,15 +12,9 @@
  */
 class Parser
 {
-    /**
-     * @var YAMLParser
-     */
-    private $yamlParser;
+    private YAMLParser|null $yamlParser;
 
-    /**
-     * @var MarkdownParser
-     */
-    private $markdownParser;
+    private MarkdownParser|null $markdownParser;
```

Comment:
I don't think these properties can be null, the constructor sets a default value

Reply:
I don't believe that's the case. As they're allowed to be either a specific type or null, and are set to null by default, then both should be allowed.

Reply:
Look here: https://github.com/mnapoli/FrontYAML/pull/32/files#diff-47bc8653b21716576958e25b6d7356ecb0f0070f17554d12d2bee985ac211b26R33-R34

these properties cannot be null, they are always set a non-null value in the constructor

Reply:
Reverted.